plugins/essif-lab_contactform7/src/js/script.js

```diff
@@ -0,0 +1,25 @@
+(function( $ ) {
+	$( window ).load(function() {
+		$("#essif-lab").click(function (e) {
+			e.preventDefault();
+			sendMockCall();
+		});
+
+		function sendMockCall() {
```

Comment:
Ik stel voor dat we deze functie in een js file in de hoofd plugin definieren. Dan kunnen we vanuit alle subplugins een andere (custom) js file inladen die een dependency heeft met de js file in de hoofdplugin. In deze subplugin js file staan dan alleen de event handlers (in dit geval de on click).

Dan moet er alleen nog vanuit php een nonce gegenereerd worden die aan de hoofdplugin js functie als argument mee wordt gegeven. Deze nonce wordt vervolgens uitgelezen en vertaald door de hoofdplugin. Waaruit op wordt gemaakt om welke validation policy het gaat en dus zo dus weet welke aanvraag naar de essif glue laag verstuurd moet worden.

Dit betekend dus dat deze hoofdplugin js functie async moet zijn omdat er van js naar php gecommuniceerd wordt. Kortom dit is hoe ik het voor me zie:

1. De eindgebruiker bezoekt een pagina waarop een "target" getoond wordt.
2. Met de php van de subplugin wordt een globale js waardes op de pagina geladen die subplugin specifiek is en verteld om welke target het gaat. 
3. De eindgebruiker klikt op de "Gegevens inladen uit wallet" knop.
4. De js van de subplugin vangt het klik event op en roept de hoofd plugin js functie aan met de identifier van het desbetreffende target.
5. De hoofd plugin js functie roept een custom api aan van de hoofd plugin (die dmv php & wp opgesteld kan worden).
6. De custom api van de hoofd plugin genereert mbv de target identifier het request voor de essif glue API. Hierbij wordt de opgemaakt om welke hook/subplugin het gaat en kan de subplugin gevraagd worden om voor het desbetreffende target een callback url te genereren.
7. Het request naar de essif glue api is gegenereerd door de custom api en kan doorgestuurd naar de js die de custom api aansprak.
8. De js verstuurd het request naar de essif glue api waarbij de gebruiker dus moet worden doorgestuurd.


**Dat is een hele lap tekst en ook behoorlijk wat werk, voor nu is dit prima**.

@Ufuk09 zou je hier nieuwe kaarten voor kunnen aanmaken?